services/src/main/scala/org/apache/gearpump/services/MasterService.scala

```diff
@@ -98,14 +97,16 @@ trait MasterService extends HttpService {
       } ~
       path("submitapp") {
         post {
-          respondWithMediaType(MediaTypes.`application/json`) {
-            entity(as[MultipartFormData]) { formData =>
-              val jar = formData.fields.head.entity.data.toByteArray
-              onComplete(Future(MasterService.submitJar(jar, system.settings.config))) {
-                case Success(_) =>
-                  complete(write(MasterService.Status(true)))
-                case Failure(ex) =>
-                  failWith(ex)
+          anyParams('args.as[Option[String]]) { (extraArgs) =>
```

Comment:
I'm not sure extra args should be handled in this way versus providing them within a conf file (that would be packaged in the assembly). It would be difficult to handle the various types of CLI args possible including those that override specific conf properties. We may be better off handling this within the context of jar packaging.